File: vote-admin/backend-spring/src/main/java/com/voteadmin/controller/AuthController.java
package com.voteadmin.controller;

import com.voteadmin.service.AuthService;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody LoginRequest req) {
        if (!"admin".equals(req.getUsername()) || !"admin123".equals(req.getPassword())) {
            return ResponseEntity.status(401).body(Map.of("error", "Invalid credentials"));
        }
        String token = authService.generateToken("admin-001", "ADMIN");
        return ResponseEntity.ok(Map.of(
                "success", true,
                "token", token,
                "role", "ADMIN"
        ));
    }

    @Data
    public static class LoginRequest {
        private String username;
        private String password;
    }
}

File: vote-admin/backend/src/controllers/authController.js
import asyncHandler from 'express-async-handler';
import jwt from 'jsonwebtoken';

const JWT_SECRET = process.env.JWT_SECRET || 'dev_jwt_secret';

/**
 * @desc    Admin login (temporary / demo-safe)
 * @route   POST /api/auth/login
 */
export const adminLogin = asyncHandler(async (req, res) => {
  const { username, password } = req.body;

  // 🔒 DEMO LOGIN (for project / viva)
  if (username !== 'admin' || password !== 'admin123') {
    res.status(401);
    throw new Error('Invalid credentials');
  }

  const token = jwt.sign(
    {
      id: 'admin-001',
      role: 'ADMIN',
    },
    JWT_SECRET,
    { expiresIn: '2h' }
  );

  res.json({
    success: true,
    token,
    role: 'ADMIN',
  });
});
